core/src/main/java/org/elasticsearch/index/mapper/fixed/FixedPointFieldMapper.java

```diff
@@ -0,0 +1,399 @@
+            long maxValue = NumericUtils.getMaxLong(terms);
+            return new FieldStats.Long(
+                maxDoc, terms.getDocCount(), terms.getSumDocFreq(), terms.getSumTotalTermFreq(), minValue, maxValue
+            );
+        }
+    }
+
+    protected FixedPointFieldMapper(String simpleName, MappedFieldType fieldType, MappedFieldType defaultFieldType,
+                              Explicit<Boolean> ignoreMalformed, Explicit<Boolean> coerce,
+                              Settings indexSettings, MultiFields multiFields, CopyTo copyTo) {
+        super(simpleName, fieldType, defaultFieldType, ignoreMalformed, coerce, indexSettings, multiFields, copyTo);
+    }
+
+    @Override
+    public FixedPointFieldType fieldType() {
+        return (FixedPointFieldType) super.fieldType();
+    }
+
+    @Override
+    protected boolean customBoost() {
+        return true;
+    }
+
+    @Override
+    protected void innerParseCreateField(ParseContext context, List<Field> fields) throws IOException {
+        double value;
+        float boost = fieldType().boost();
+        if (context.externalValueSet()) {
+            Object externalValue = context.externalValue();
+            if (externalValue == null) {
+                if (fieldType().nullValue() == null) {
+                    return;
+                }
+                value = fieldType().nullValue();
+            } else if (externalValue instanceof String) {
+                String sExternalValue = (String) externalValue;
+                if (sExternalValue.length() == 0) {
+                    if (fieldType().nullValue() == null) {
+                        return;
+                    }
+                    value = fieldType().nullValue();
+                } else {
+                    value = Double.parseDouble(sExternalValue);
+                }
+            } else {
+                value = ((Number) externalValue).doubleValue();
+            }
+            if (context.includeInAll(includeInAll, this)) {
+                context.allEntries().addText(fieldType().name(), Double.toString(value), boost);
+            }
+        } else {
+            XContentParser parser = context.parser();
+            if (parser.currentToken() == XContentParser.Token.VALUE_NULL ||
+                (parser.currentToken() == XContentParser.Token.VALUE_STRING && parser.textLength() == 0)) {
+                if (fieldType().nullValue() == null) {
+                    return;
+                }
+                value = fieldType().nullValue();
+                if (fieldType().nullValueAsString() != null && (context.includeInAll(includeInAll, this))) {
+                    context.allEntries().addText(fieldType().name(), fieldType().nullValueAsString(), boost);
+                }
+            } else if (parser.currentToken() == XContentParser.Token.START_OBJECT
+                && Version.indexCreated(context.indexSettings()).before(Version.V_3_0_0)) {
```

Comment:
We should remove this block even if we plan to backport to 2.x. This is a deprecated feature that allows to provide values as an object so that you can also pas a boost in. We should not add this deprecated feature to a new field type.